osu.Game/Screens/OnlinePlay/Multiplayer/Match/MatchStartControl.cs

```diff
@@ -101,16 +105,31 @@ protected override void OnRoomLoadRequested()
             endOperation();
         }
 
-        private void onReadyClick()
+        private void onReadyButtonClick()
         {
             if (Room == null)
                 return;
 
             Debug.Assert(clickOperation == null);
             clickOperation = ongoingOperationTracker.BeginOperation();
 
-            if (isReady() && Client.IsHost && !Room.ActiveCountdowns.Any(c => c is MatchStartCountdown))
-                startMatch();
+            if (Client.IsHost)
+            {
+                if (Room.State == MultiplayerRoomState.Open)
+                {
+                    if (isReady() && !Room.ActiveCountdowns.Any(c => c is MatchStartCountdown))
+                        startMatch();
+                    else
+                        toggleReady();
+                }
+                else
```

Comment:
Given that the other condition checks room state, I'd probably have this `else` check that it's not `MultiplayerRoomStata.Closed` just for safety.

Reply:
I guess... Nothing's really being achieved here in my eyes, not even with respect to safety.